plugins/dell-kestrel/fu-dell-kestrel-package.c

```diff
@@ -66,20 +64,12 @@ fu_dell_kestrel_package_write(FuDevice *device,
 	fw = fu_firmware_get_bytes(firmware, error);
 	if (fw == NULL)
 		return FALSE;
-	data = g_bytes_get_data(fw, &length);
-	if (!fu_memcpy_safe((guint8 *)&status_version,
-			    sizeof(status_version),
-			    0x0,
-			    data,
-			    length,
-			    FU_DELL_KESTREL_PACKAGE_VERSION_OFFSET,
-			    sizeof(status_version),
-			    error))
-		return FALSE;
+	data = g_bytes_get_data(fw, &data_sz);
 
 	/* new version */
-	dynamic_version =
-	    fu_version_from_uint32_hex(status_version, fu_device_get_version_format(device));
+	if (!fu_memread_uint32_safe(data, data_sz, 0x0, &buf_pkg, G_LITTLE_ENDIAN, error))
```

Comment:
I'd much rather we use `fu_firmware_get_stream()` and `fu_input_stream_read_u32()` here -- it's way more efficient than reading out the entire buffer.

Reply:
fixed